### ISSUE TITLE
Could C++ symbols be demangled?

### ISSUE THREAD

**Issue by anntzer** [vmprof/vmprof-python#150]
All's in the title, I believe...
Thanks!

**Comment by planrich** [comment]
Yes could be done. Though it would need a special demangling for each compiler.

```
class Test {
    public:
        Test();
        void foo();
}
```

If you compile a cpp file to an object file and lookup the symbol it would return `_ZN4Test3fooEv` for the function `void foo()`. This is different for each compiler. There might be demangling tools that we could use. Patches welcome!
